### ISSUE TITLE
Alternative options on a symbolic fd 

### ISSUE THREAD

**Issue by Usibre** [angr/angr#1195]
Whenever a symbolic file descriptor (fd) is read, a warning is logged: 

> WARNING | 2018-09-11 16:25:56,893 | angr.state_plugins.posix | Tried to look up a symbolic fd - constrained to 4 and opened /tmp/angr_implicit_0

it automatically chooses an unused file descriptor, even if a used file descriptor is available. I personally encounter the symbolic fd after a bug, and I was hoping there would be a choice of using an existing fd (or for example a branching with (1) unused fd and (2) existent fd's)

This would be a nice feature to have! 

Thanks!

**Comment by rhelmot** [comment]
You're right! This would be cool. Something like concretization strategies, but for file descriptors. The easiest verison of this would just take a parameter to the posix plugin, then call it whenever we would otherwise open an implicit file.
